grpc/swift-diatheke/Client.swift

```diff
@@ -66,185 +66,203 @@ public extension Cobaltspeech_Diatheke_DiathekeClient {
         self.init(channel: connection)
     }
     
-    func diathekeVersion(success: @escaping (_ version: String) -> (), failure: DiathekeFailureCallback?) {
+    // Version returns version information from the server.
+    func version(success: @escaping (_ versionResponse: Cobaltspeech_Diatheke_VersionResponse) -> (),
+                 failure: @escaping DiathekeFailureCallback) {
         version(Cobaltspeech_Diatheke_Empty()).response.whenComplete { (result) in
             switch result {
-            case .success(let response):
-                success(response.server)
+            case .success(let versionResponse):
+                success(versionResponse)
             case .failure(let error):
-                failure?(error)
+                failure(error)
             }
         }
     }
     
-    func listModels(success: @escaping (_ models: [String]) -> (), failure: DiathekeFailureCallback?) {
-        models(Cobaltspeech_Diatheke_Empty()).response.whenComplete { (result) in
+    // ListModels returns a list of Diatheke models available to the server.
+    func listModels(success: @escaping (_ models: [Cobaltspeech_Diatheke_ModelInfo]) -> (),
+                    failure: @escaping DiathekeFailureCallback) {
+        listModels(Cobaltspeech_Diatheke_Empty()).response.whenComplete { (result) in
             switch result {
             case .success(let response):
                 success(response.models)
             case .failure(let error):
-                failure?(error)
+                failure(error)
             }
         }
     }
     
-    func newSession(model: String,
-                    success: @escaping (_ sessionID: String) -> (),
-                    failure: DiathekeFailureCallback?) {
-        var request = Cobaltspeech_Diatheke_NewSessionRequest()
-        request.model = model
-        
-        newSession(request).response.whenComplete { (result) in
+    // CreateSession creates a new Diatheke session using the specified model.
+    func createSession(modelID: String,
+                       success: @escaping (Cobaltspeech_Diatheke_SessionOutput) -> (),
```

Comment:
success is a function that we will call, passing in the session output, but what does the `@escaping` keyword mean?

Reply:
> success is a function that we will call, passing in the session output, but what does the `@escaping` keyword mean?

`@escaping` says that `success` function will be called outside of the `createSession` function.